odl/util/numerics.py

```diff
@@ -297,6 +303,593 @@ def fast_1d_tensor_mult(ndarr, onedim_arrs, axes=None, out=None):
+                 pad_mode='constant', pad_const=0, direction='forward',
+                 out=None):
+    """Return the resized version of ``arr`` with shape ``newshp``.
+
+    In axes where ``newshp > arr.shape``, padding is applied according
+    to the supplied options.
+    Where ``newshp < arr.shape``, the array is cropped to the new
+    size.
+
+    Parameters
+    ----------
+    arr : array-like
+        Array to be resized.
+    newshp : sequence of int
+        Desired shape of the output array.
+    frac_left : float or sequence of float, optional
+        Number between 0 and 1 (inclusive) that determines which fraction
+        of the addition/removal is performed on the "left" side. If
+        ``frac_left`` times the shape difference is not integer, the
+        number of entries added to the left is gained by rounding to the
+        nearest integer. The default ``None`` is equivalent to 0.5.
+        This option can be specified per axis with a sequence.
+        Cannot be combined with ``num_left``.
+    num_left : sequence of int, optional
+        Specifies how many entries are added to/removed from the left
+        side of ``arr``.
+        Cannot be combined with ``frac_left``.
+    pad_mode : str, optional
+        Method to be used to fill in missing values in an enlarged array.
+
+        ``'constant'``: Fill with ``pad_const``.
+
+        ``'symmetric'``: Reflect at the boundaries, not doubling the
+        outmost values. This requires left and right padding sizes
+        to be strictly smaller than the original array shape.
+
+        ``'periodic'``: Fill in values from the other side, keeping
+        the order. This requires left and right padding sizes to be
+        at most as large as the original array shape.
+
+        ``'order0'``: Extend constantly with the outmost values
+        (ensures continuity).
+
+        ``'order1'``: Extend with constant slope (ensures continuity of
+        the first derivative). This requires at least 2 values along
+        each axis where padding is applied.
+
+        Note that for ``'symmetric'`` and ``'periodic'`` padding, the
+        number of added values on each side of the array cannot exceed
+        the original size.
+
+    pad_const : scalar, optional
+        Value to be used in the ``'constant'`` padding mode.
+    direction : {'forward', 'adjoint'}
+        Determines which variant of the resizing is applied.
+
+        'forward' : in axes where ``out`` is larger than ``arr``,
+        apply padding. Otherwise, restrict to the smaller size.
+
+        'adjoint' : in axes where ``out`` is larger than ``arr``,
+        apply zero-padding. Otherwise, restrict to the smaller size
+        and add the outside contributions according to ``pad_mode``.
+
+    out : `numpy.ndarray`, optional
+        Array to write the result to. Must have shape ``newshp`` and
+        be able to hold the data type of the input array.
```

Comment:
None needed? We assign stuff by slicing, that will work irrespective of the data order. Performance will depend on it, sure.